src/form-builder/reducers/controlDetails.js

```diff
@@ -1,18 +1,41 @@
 import { cloneDeep } from 'lodash';
 const controlDetails = (store = {}, action) => {
   switch (action.type) {
+    // eslint-disable-next-line no-case-declarations
     case 'SELECT_CONTROL':
-      return Object.assign({}, store, { selectedControl: action.metadata });
+      let storeClone;
+      if (store.allControls === undefined) {
+        storeClone = Object.assign({}, store, { allControls: [{ id: action.metadata.id,
+          name: action.metadata.concept.name, events: action.metadata.events }] });
+      } else {
+        storeClone = store;
+        storeClone.allControls = storeClone.allControls.filter(control =>
+          control.id !== action.metadata.id);
+        storeClone.allControls = storeClone.allControls.concat({ id: action.metadata.id,
+          name: action.metadata.concept !== undefined ? action.metadata.concept.name : undefined,
+          events: action.metadata.events });
+      }
+      return Object.assign({}, storeClone, { selectedControl: action.metadata });
     case 'DESELECT_CONTROL':
       return Object.assign({}, store, { selectedControl: undefined });
     case 'FOCUS_CONTROL':
       return Object.assign({}, store, { focusedControl: action.id });
     case 'BLUR_CONTROL':
       return Object.assign({}, store, { focusedControl: undefined });
     case 'SOURCE_CHANGED':
-      // eslint-disable-next-line
+      // eslint-disable-next-line no-param-reassign
       store.selectedControl.events = { onValueChange: action.source };
+      // eslint-disable-next-line no-param-reassign
+      store.allControls = store.allControls.map(control =>
+        (control.id === action.id ?
+          Object.assign({}, control, { events: { onValueChange: action.source } })
+          : control)
+      );
       return cloneDeep(store);
+    case 'FORM_LOAD':
+      return Object.assign({}, store, { allControls:
```

Comment:
Should the property be called allControlEvents?

Reply:
I renamed it to allObsControlEvents